### PR TITLE
Add `--force` flag to `tsh scp` to allow overwriting read only files

### PR DESCRIPTION
**Pull request by lxea**
Fix for #21191

### CODE REVIEW

File: lib/sshutils/sftp/remote.go [hunk 1]
```diff
@@ -83,12 +83,23 @@ func (r *remoteFS) Create(ctx context.Context, path string, mode os.FileMode) (i
 
 	f, err := r.c.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, mode)
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return nil, trace.ConvertSystemError(err)
```
Comment (codingllama) [inline lib/sshutils/sftp/remote.go:86]: Should the remoteFS keep the trace.Wrap? Do we really see system errors here?

File: lib/sshutils/sftp/remote.go [hunk 2]
```diff
@@ -83,12 +83,23 @@ func (r *remoteFS) Create(ctx context.Context, path string, mode os.FileMode) (i
...
 	return f, nil
 }
 
+func (r *remoteFS) Remove(ctx context.Context, path string) error {
+	if err := ctx.Err(); err != nil {
+		return err
+	}
+
+	if err := r.c.Remove(path); err != nil {
+		return trace.ConvertSystemError(err)
```
Comment (codingllama) [inline lib/sshutils/sftp/remote.go:98]: Ditto for trace.Wrap?

File: lib/sshutils/sftp/sftp_test.go [hunk 3]
```diff
@@ -204,6 +247,16 @@ func TestUpload(t *testing.T) {
 			} else {
 				require.EqualError(t, err, fmt.Sprintf(tt.expectedErr, tempDir))
 			}
+
+			if tt.tryOverwrite {
```
Comment (codingllama) [inline lib/sshutils/sftp/sftp_test.go:251]: I don't follow why we need another block, nor why the above doesn't error. Could the cfg.transfer above run this scenario?

**Comment by codingllama** [comment]
Looks good, holding out for that one testing Q.

File: lib/sshutils/sftp/sftp_test.go (hunk 3 above)
Comment (camscale) [reply lib/sshutils/sftp/sftp_test.go:251]: Agree with @codingllama here - this block seems to just repeat the same test without changing anything, and I am confused how the tests pass when `tt.expectedErr != tt.expectedOverwriteErr` - it looks like the first time the tests run, it should fail the test as `tt.expectedErr == ""`.

File: lib/sshutils/sftp/sftp.go [hunk 4]
```diff
@@ -388,15 +389,13 @@ func (c *Config) transferFile(ctx context.Context, dstPath, srcPath string, srcF
...
 			if err := c.dstFS.Remove(ctx, dstPath); err != nil {
 				return trace.Wrap(err)
 			}
 			dstFile, err = c.dstFS.Create(ctx, dstPath, srcFileInfo.Mode())
-			if err != nil {
-				return trace.Wrap(err)
-			}
-		} else {
+		} // Re-check error as it may be nil from the remove-create case above.
+		if err != nil {
```
Comment (codingllama) [inline lib/sshutils/sftp/sftp.go:398]: I think I prefer the original block inside the if/right after the Create, but up to you.

File: lib/sshutils/sftp/sftp_test.go (hunk 3 above)
Comment (codingllama) [reply lib/sshutils/sftp/sftp_test.go:251]: Mind adding a comment? Doesn't have to be long, but having something would help.

**Review by codingllama — APPROVED** [review]

File: lib/sshutils/sftp/sftp_test.go (hunk 3 above)
Comment (camscale) [reply lib/sshutils/sftp/sftp_test.go:251]: Thanks. I saw that the file mode was specified as 0o440, but it didn't click that it was in the src location and that you first needed to copy it with that mode to the destination, and then do it again to actually test the new logic.

File: lib/sshutils/sftp/sftp.go (hunk 4 above)
Comment (camscale) [reply lib/sshutils/sftp/sftp.go:398]: Oh dear. Sorry @lxea - duelling reviewers are no fun. Mine was just a suggestion, so likewise - up to you.

**Review by camscale — APPROVED** [review]

**Comment by r0mant** [comment]
@lxea What's the status of this PR? Can you rebase and merge?

Also, can you document this new flag in this PR as well: https://goteleport.com/docs/reference/cli/#tsh-scp.

**Comment by zmb3** [comment]
@lxea haven't heard from you on this one after multiple pings, so I'm going to close it out.

**Pull request closed without merging**